### PR TITLE
Fix masking for torch

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/torch/distributions.py [hunk 1]
```diff
@@ -186,11 +186,13 @@ def _create_policy_branches(self, hidden_size: int) -> nn.ModuleList:
...
-    def _mask_branch(self, logits: torch.Tensor, mask: torch.Tensor) -> torch.Tensor:
+    def _mask_branch(
+        self, logits: torch.Tensor, allow_mask: torch.Tensor
+    ) -> torch.Tensor:
         # Zero out masked logits, then subtract a large value. Technique mentionend here:
-        # https://arxiv.org/abs/2006.14171. Our implementation is ONNX and Barrcuda-friendly.
-        flipped_mask = 1.0 - mask
-        adj_logits = logits * mask - 1e8 * flipped_mask
+        # https://arxiv.org/abs/2006.14171. Our implementation is ONNX and Barracuda-friendly.
+        mask = -1.0 * allow_mask + 1.0
```
Comment (ervteng) [inline ml-agents/mlagents/trainers/torch/distributions.py:194]: maybe call this `block_mask`?

**Review by ervteng — APPROVED** [review]
Python changes look good

File: ml-agents/mlagents/trainers/torch/distributions.py (hunk 1 above)
Comment (chriselion) [inline ml-agents/mlagents/trainers/torch/distributions.py:194]: Please add a comment that the `-1 * x + 1` format is to workaround a bug, so that someone doesn't "optimize" it in the future.

**Review by dongruoping — APPROVED** [review]
Second on Ervin's comments, otherwise LGTM
